### PR TITLE
Prepare repo for 4.9 freeze

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Cheery-pick changes from upstream, and accommodate Operator for last NFD changes

- Code optimization (remove not needed it/else checks)
- Delete validateUpdateEvent func, not used currently, left as TODO once we have more SRE experience on running NFD
- Add support for config IMAGE_TAG and rbac proxy image
- Add support for extraLabelNs, resourceLabels, labelWhiteList
- Fix rbac kustomize location and uniq project dir
- Makefile: make BUNDLE_IMG env var sensible 
- Delete updateAvailableCondition func, not needed since we straight update the CR using controller-runtime

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -60,13 +62,16 @@ PACKAGE=github.com/openshift/cluster-nfd-operator
 MAIN_PACKAGE=main.go
 BIN=node-feature-discovery-operator
 
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+
 all: build
 
+# Run tests
```
Comment (dagrayvid) [inline Makefile:69]:
```suggestion
```

File: controllers/nodefeaturediscovery_controller.go [hunk 2]
```diff
@@ -479,21 +455,6 @@ func (r *NodeFeatureDiscoveryReconciler) SetupWithManager(mgr ctrl.Manager) erro
 		Complete(r)
 }
 
-// validateUpdateEvent validates whether or not an NFD Operator resource has
-// an update.
-func validateUpdateEvent(e *event.UpdateEvent) bool {
```
Comment (dagrayvid) [inline controllers/nodefeaturediscovery_controller.go:484]: Can you add to the PR description an explanation for deleting this function?

Comment (ArangoGutierrez) [reply controllers/nodefeaturediscovery_controller.go:484]: Done

File: config/samples/nfd.openshift.io_v1_nodefeaturediscovery.yaml [hunk 3]
```diff
@@ -24,8 +31,9 @@ spec:
       #    stderrthreshold: 2
       #    v: 0
       #    vmodule:
-      ##   NOTE: the following options are not dynamically run-time configurable
-      ##         and require a nfd-worker restart to take effect after being changed
+      ##   NOTE: the following options are not dynamically run-time 
+      ##          Configurable and require a nfd-worker restart to take effect
+      ##          after being changed
```
Comment (dagrayvid) [inline config/samples/nfd.openshift.io_v1_nodefeaturediscovery.yaml:36]:
```suggestion
      ##   NOTE: the following options are not dynamically run-time 
      ##          configurable and require a nfd-worker restart to take effect
      ##          after being changed
```

File: controllers/nodefeaturediscovery_status.go [hunk 4]
```diff
@@ -156,17 +151,6 @@ func (r *NodeFeatureDiscoveryReconciler) updateProgressingCondition(nfd *nfdv1.N
 	return reconcile.Result{Requeue: true}, nil
 }
 
-// updateAvailableCondition is used to mark a given resource as "progressing" so
-// that the reconciler can take steps to rectify the situation.
-func (r *NodeFeatureDiscoveryReconciler) updateAvailableCondition(nfd *nfdv1.NodeFeatureDiscovery) (ctrl.Result, error) {
```
Comment (dagrayvid) [inline controllers/nodefeaturediscovery_status.go:161]: Can you explain the deletion of this function in the PR description?

**Comment by dagrayvid** [comment]
Thanks for addressing all of my comments @ArangoGutierrez 

/lgtm

**Review by lenahorsley — COMMENTED** [review]
/lgtm

**Comment by lenahorsley** [comment]
> /lgtm

Tested 3 September 2021
Created and deployed a bundle to an AWS cluster.
All worker nodes were labeled correctly.

**Pull request merged**